src/connectors/infinite-hits/__tests__/connectInfiniteHits-test.ts

```diff
@@ -968,6 +968,134 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/infinite-hi
     });
   });
 
+  describe('getRenderState', () => {
+    it('returns the render state', () => {
+      const renderFn = jest.fn();
+      const unmountFn = jest.fn();
+      const createInfiniteHits = connectInfiniteHits(renderFn, unmountFn);
+      const infiniteHitsWidget = createInfiniteHits({});
+      const helper = algoliasearchHelper(createSearchClient(), 'indexName', {
+        index: 'indexName',
+      });
+
+      const renderState1 = infiniteHitsWidget.getRenderState!(
+        {},
+        createInitOptions({ state: helper.state, helper })
+      );
+
+      expect(renderState1.infiniteHits).toEqual({
+        hits: [],
+        sendEvent: expect.any(Function),
+        bindEvent: expect.any(Function),
+        isFirstPage: true,
+        isLastPage: true,
+        results: undefined,
+        showMore: expect.any(Function),
+        showPrevious: expect.any(Function),
+        widgetParams: {},
+      });
+
+      const hits = [
+        { objectID: '1', name: 'name 1' },
+        { objectID: '2', name: 'name 2' },
+      ];
+
+      const results = new SearchResults(helper.state, [
+        createSingleSearchResponse({ hits, queryID: 'theQueryID' }),
+      ]);
+
+      const renderState2 = infiniteHitsWidget.getRenderState!(
```

Comment:
this can be two separate tests if the second uses `getInitializedWidget`

Reply:
I don't really see the advantage of doing this, we will still have 2 renders to do in the second test anyway. Also since there is a relation between the 2 renders (the functions we got from the second render must match the ones we received in the first), I think it is clearer to keep everything in the same test.